From 986c9fbed6cb65a46925710534691f0dc644c34d Mon Sep 17 00:00:00 2001
From: David S. Wang <dsw@cloudera.com>
Date: Wed, 27 Mar 2013 16:48:08 -0700
Subject: [PATCH 42/96] HBASE-8067 TestHFileArchiving.testArchiveOnTableDelete sometimes fails -- more debug

Reason: Debug
Author: Matteo Bertozzi
Ref: CDH-10058
(cherry picked from commit 8da27dea27b03c8dc497e74f180140162ac6ae1d)
---
 .../hbase/master/handler/DeleteTableHandler.java   |    4 ++++
 .../hadoop/hbase/backup/TestHFileArchiving.java    |    1 +
 2 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/src/main/java/org/apache/hadoop/hbase/master/handler/DeleteTableHandler.java b/src/main/java/org/apache/hadoop/hbase/master/handler/DeleteTableHandler.java
index 2781671..957e531 100644
--- a/src/main/java/org/apache/hadoop/hbase/master/handler/DeleteTableHandler.java
+++ b/src/main/java/org/apache/hadoop/hbase/master/handler/DeleteTableHandler.java
@@ -92,11 +92,15 @@ public class DeleteTableHandler extends TableEventHandler {
       if (!fs.delete(tempTableDir, true)) {
         LOG.error("Couldn't delete " + tempTableDir);
       }
+
+      LOG.debug("Table '" + Bytes.toString(tableName) + "' archived!");
     } finally {
       // 6. Update table descriptor cache
+      LOG.debug("Removing '" + Bytes.toString(tableName) + "' descriptor.");
       this.masterServices.getTableDescriptors().remove(Bytes.toString(tableName));
 
       // 7. If entry for this table in zk, and up in AssignmentManager, remove it.
+      LOG.debug("Marking '" + Bytes.toString(tableName) + "' as deleted.");
       am.getZKTable().setDeletedTable(Bytes.toString(tableName));
     }
   }
diff --git a/src/test/java/org/apache/hadoop/hbase/backup/TestHFileArchiving.java b/src/test/java/org/apache/hadoop/hbase/backup/TestHFileArchiving.java
index c2b3d76..5d27a98 100644
--- a/src/test/java/org/apache/hadoop/hbase/backup/TestHFileArchiving.java
+++ b/src/test/java/org/apache/hadoop/hbase/backup/TestHFileArchiving.java
@@ -244,6 +244,7 @@ public class TestHFileArchiving {
 
     // then delete the table so the hfiles get archived
     UTIL.deleteTable(TABLE_NAME);
+    LOG.debug("Deleted table");
 
     // then get the files in the archive directory.
     Path archiveDir = HFileArchiveUtil.getArchivePath(UTIL.getConfiguration());
-- 
1.7.0.4

